libs/agno/agno/tools/openai.py

```diff
@@ -36,3 +46,69 @@ def transcribe_audio(self, audio_path: str) -> str:
         except Exception as e:
             logger.error(f"Failed to transcribe audio: {str(e)}")
             return f"Failed to transcribe audio: {str(e)}"
+
+    def generate_image(self, prompt: str, n: int = 1, size: OpenAIImageSize = "1024x1024") -> str:
+        """Generate images based on a text prompt using OpenAI's DALL-E API
+        Args:
+            prompt (str): The text prompt to generate the image from.
+            n (int): The number of images to generate. Defaults to 1. Must be 1 for dall-e-3.
+            size (OpenAIImageSize): The size of the generated images. Defaults to "1024x1024".
+                Must be one of "1024x1024", "1792x1024", or "1024x1792" for dall-e-3.
+                (Also supports "256x256", "512x512" for dall-e-2).
+
+        Returns:
+            str: A comma-separated string of URLs pointing to the generated images, or an error message.
+        """
+        log_info(f"Generating {n} image(s) of size {size} for prompt: '{prompt}'")
+        try:
+            # Actual API call
+            response = client.images.generate(
+                model="dall-e-3",  # Use dall-e-3
+                prompt=prompt,
+                n=n,  # Note: DALL-E 3 currently only supports n=1
+                size=size,
+                response_format="url",  # Request URLs instead of base64 data
+            )
+            image_urls = [item.url for item in response.data if item.url]  # Extract URLs
+            result_string = ", ".join(image_urls)  # Convert list to comma-separated string
+            log_info(f"Generated image URLs (as string): {result_string}")
+            return result_string  # Return string instead of list
+        except Exception as e:
+            logger.error(f"Failed to generate image: {str(e)}")
+            # Consider more specific error handling based on OpenAI exceptions if needed
+            # e.g., from openai import APIError, RateLimitError, BadRequestError etc.
+            return f"Failed to generate image: {str(e)}"  # Return error message as string
+
+    def generate_speech(
+        self,
+        text_input: str,
+        output_path: str,
```

Comment:
Not sure about this... we actually only want the audio artifact, not automatically stored to a file? We have utils the the user can use to write to file, but otherwise if it has to be here, we should set the output path on init